### PR TITLE
added initial hometask implementation

### PR DESCRIPTION
**Pull request by RossKostenko**
Added initial hometask implementation

### CODE REVIEW

File: hometask-1-OOP/src/Point.ts [hunk 1]
```diff
@@ -1 +1,53 @@
...
+			const y = this.y - args[1];
+
+			return this.calculateDistance(x, y);
+		}
+
+		throw new Error('OOPS, Something went wrong in distance method');
+	}
+
+	private calculateDistance(x: number, y: number): number {
+		return Math.round(Math.sqrt(Math.pow(x, 2) + Math.pow(y, 2)) * 10) / 10;
```
Comment (dctenna) [inline hometask-1-OOP/src/Point.ts]: nit: we have Math.hypot(x, y) method to calculate hypotenuse. No need to fix this comment. Just FYI)

File: hometask-1-OOP/src/Shape.ts [hunk 2]
```diff
@@ -1,3 +1,58 @@
...
+
+		this.points = points;
+		this.color = color;
+		this.filled = filled;
+	}
+
+	abstract getType(): string;
+
+	public toString() {
+		const pointsStringified = this.points.map((point: Point): string => point.toString()).join(', ');
```
Comment (dctenna) [inline hometask-1-OOP/src/Shape.ts:24]: you can move this logic to a private method.

File: hometask-1-OOP/src/Triangle.ts [hunk 3]
```diff
@@ -1,3 +1,51 @@
...
+		if (color && typeof filled === 'boolean') {
+			super(points, color, filled);
+			return;
+		}
+
+		super(points);
+	}
+
+	public toString(): any {
+		const pointsStringified = this.points.map((point: Point, idx: number): string => `v${idx + 1}=${point.toString()}`).join(',');
```
Comment (dctenna) [inline hometask-1-OOP/src/Triangle.ts:20]: you can move this logic to a private method.

**Review by dctenna — APPROVED** [review]

File: hometask-1-OOP/src/Point.ts (hunk 1 above)
Comment (RossKostenko) [reply hometask-1-OOP/src/Point.ts]: Oh, thanks, didn't know about that!

File: hometask-1-OOP/src/Shape.ts (hunk 2 above)
Comment (RossKostenko) [reply hometask-1-OOP/src/Shape.ts:24]: yup, thanks. 

Could there be any issues regarding of inheritance of private methods from abstract class.

File: hometask-1-OOP/src/Triangle.ts (hunk 3 above)
Comment (RossKostenko) [reply hometask-1-OOP/src/Triangle.ts:20]: Thanks!

**Pull request merged**